src/RadixDlt.NetworkGateway.PostgresIntegration/ScryptoSborUtils.cs

```diff
@@ -112,61 +119,121 @@ public static GatewayModel.EntityMetadataItemValue MetadataValueToGatewayMetadat
+            asString: asString,
+            asStringCollection: asStringCollection);
+    }
+
+    public static string GetSimpleStringOfMetadataValue(Enum variantEnum)
+    {
+        switch (variantEnum.Variant)
+        {
+            // See https://github.com/radixdlt/radixdlt-scrypto/blob/release/rcnet-v1/transaction/examples/metadata/metadata.rtm
+            case 0 when variantEnum.Fields is [String value]:
+                return value.Value;
+            case 1 when variantEnum.Fields is [Bool value]:
+                return value.Value.ToString();
+            case 2 when variantEnum.Fields is [U8 value]:
+                return value.Value.ToString();
+            case 3 when variantEnum.Fields is [U32 value]:
+                return value.Value.ToString();
+            case 4 when variantEnum.Fields is [U64 value]:
+                return value.Value.ToString();
+            case 5 when variantEnum.Fields is [I32 value]:
+                return value.Value.ToString();
+            case 6 when variantEnum.Fields is [I64 value]:
+                return value.Value.ToString();
+            case 7 when variantEnum.Fields is [Decimal value]:
+                return value.Value;
+            case 8 when variantEnum.Fields is [Address value]:
+                return value.TmpAddress;
+            case 9 when variantEnum.Fields is [Enum publicKeyEnum]:
+                var keyName = publicKeyEnum.Variant switch
                 {
-                    var isValid = true;
-                    var asStringCollectionCandidate = new List<string>();
+                    0 => "EcdsaSecp256k1PublicKey",
+                    1 => "EddsaEd25519PublicKey",
+                    _ => $"PublicKeyType[{publicKeyEnum.Variant}]", // Fallback
+                };
 
-                    foreach (var innerEnum in innerArray.Elements.Cast<ToolkitModel.Value.ScryptoSbor.Enum>())
+                if (publicKeyEnum.Fields is [Array keyBytes])
+                {
+                    try
                     {
-                        if (innerEnum.Variant == 0 && innerEnum.Fields is [ToolkitModel.Value.ScryptoSbor.String value])
-                        {
-                            asStringCollectionCandidate.Add(value.Value);
-                        }
-                        else
-                        {
-                            isValid = false;
-                            break;
-                        }
+                        var bytes = keyBytes.Elements.Cast<U8>().Select(byteValue => byteValue.Value).ToArray();
+                        return $"{keyName}(\"{Convert.ToHexString(bytes).ToLowerInvariant()}\")";
                     }
-
-                    if (isValid)
+                    catch (InvalidCastException)
                     {
-                        asStringCollection = asStringCollectionCandidate;
+                        // Fallthrough to default
```

Comment:
if we intend to ignore it could we just log exception to improve observability and debugging?